utils/python/CIME/batch_utils.py

```diff
@@ -12,29 +12,25 @@
 
 class BatchUtils(object):
 
-    def __init__(self, job, case=None, prereq_jobid=None):
+    def __init__(self, job, case=None, prereq_jobid=None, no_batch=False):
         """
         Class constructor.  We need to know where in the filesystem we are,
         so caseroot, case, machroot, machine, cimeroot
         prereq_jobid allows the user to set a prereq to the first job in this workflow
         """
         self.case = case if case is not None else Case()
         self.job = job
-        self.batchobj = None
         self.override_node_count = None
         self.batchmaker = None
         self.jobs = []
         self.prereq_jobid = prereq_jobid
 
-    def submit_jobs(self, no_batch=False):
-        if no_batch:
```

Comment:
The old code was not safe. If no_batch is changed between calls, self.batchobj would not be updated.